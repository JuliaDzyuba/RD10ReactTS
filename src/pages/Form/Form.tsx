import React, { useEffect, useRef, useContext } from 'react';
import { ThemeContext } from '../../App';
import styles from './styles.module.scss';

function Form() {
  const inputRef = useRef<HTMLInputElement>(null);
  const themeContext = useContext(ThemeContext);

  useEffect(() => {
    inputRef.current?.focus();
  },[]);

  return (    
    <div className={`container ${themeContext.theme}`}>
      <div className={styles.formWrapper}>
        <h2>Form</h2>
        <label htmlFor="input" className={styles.focusedInput}>
          <input ref={inputRef} type="text" placeholder='Enter something...'/>
        </label>
        </div>
    </div>
  );
}

export default Form;
