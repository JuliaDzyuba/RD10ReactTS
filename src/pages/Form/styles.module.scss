.formWrapper {
  height: 100vh;
}

.focusedInput input {
  outline-color: blue;
  display: block;
  margin: 30px auto;
}
