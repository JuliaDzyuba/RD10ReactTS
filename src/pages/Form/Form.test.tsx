import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Form from './Form';

const mockUsers = [{
  name: `new user`,
      currentRole: 'user',
      avatar: 'https://via.placeholder.com/300',
      summary: '',
      education: [],
      experience: [],
      skills: {
        mainStack: ['rr'],
        database: ['rr'],
        additionally: ['rr'],
        englishLevel: 'rr',
      }
}];

describe("Form component", () => {
  test('should render', () => {
    const { container } = render(<Form users={mockUsers}/>);

    expect(container).toMatchSnapshot();
  });

  test('should contain inputs', () => {
    render(<Form users={mockUsers}/>);
    const inputs = screen.getAllByRole('textbox');

    expect(inputs.length).toEqual(2);
  });

  test('should change input value if user types', () => {
    render(<Form users={mockUsers}/>);

    const input = screen.getByPlaceholderText(/Second input\.\.\./i);
    fireEvent.change(input, {target: {value: 'Some text'}});

    expect(input).toHaveValue('Some text');
  });
});
