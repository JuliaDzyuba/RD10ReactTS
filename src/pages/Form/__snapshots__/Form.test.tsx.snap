// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component should render 1`] = `
<div>
  <div
    class="container light"
  >
    <div
      class="formWrapper"
    >
      <h2>
        Form
      </h2>
      <label
        class="focusedInput"
        for="input"
      >
        <input
          placeholder="First input..."
          type="text"
        />
      </label>
      <label
        class="focusedInput"
        for="second"
      >
        <input
          id="second"
          placeholder="Second input..."
          type="text"
          value=""
        />
        <span>
          must be at least 3 characters
        </span>
      </label>
      <hr />
      <div>
        <h2>
          Users counter: 
          1
        </h2>
      </div>
    </div>
  </div>
</div>
`;
